src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -1446,6 +1455,13 @@ _public_ sd_bus* sd_bus_flush_close_unref(sd_bus *bus) {
         if (!bus)
                 return NULL;
 
+        /* Have to do this before flush() to prevent hang */
+        if (bus->busexec_pid)
+                if (kill(bus->busexec_pid, SIGTERM) == bus->busexec_pid) {
```

Comment:
hmm, we have very similar code already in sigkill_wait()... We should probably add sigterm_wait() for this use-case, and it should be much like sigkill_wait() but use SIGTERM and kill_and_sigcont() internally.